src/docbuilder/rustwide_builder.rs

```diff
@@ -781,6 +783,18 @@ impl RustwideBuilder {
         rustdoc_flags_extras.extend(UNCONDITIONAL_ARGS.iter().map(|&s| s.to_owned()));
         let cargo_args = metadata.cargo_args(&cargo_args, &rustdoc_flags_extras);
 
+        // If the explicit target is not a tier one target, we need to install it.
+        let has_build_std = cargo_args.windows(2).any(|args| {
+            args[0].starts_with("-Zbuild-std")
+                || (args[0] == "-Z" && args[1].starts_with("build-std"))
+        }) || cargo_args.last().unwrap().starts_with("-Zbuild-std");
+        if !docsrs_metadata::DEFAULT_TARGETS.contains(&target) && !has_build_std {
```

Comment:
I don't know much about `build-std` and how it's used, 

should it be documented somewhere visible that we don't add the non-tier-1-target when `build-std` is requested?

Reply:
Hmm, this is a good point. I was thinking about tier 3 targets, which don't have `std` available anyway and will fail if you try to download it, but it's allowed to use build-std for other tiers too. I'll add some documentation.

Reply:
Hmm, I think this behavior shouldn't be visible to crate authors, actually. We unconditionally run builds from an x64-linux host, which already has `std` installed, so this can't break proc-macros or build scripts. And all other dependencies will have `-Zbuild-std` applied to them, so they can't notice that `std` wasn't downloaded because it's never loaded from the sysroot.